synapse/storage/databases/main/client_ips.py

```diff
@@ -189,7 +229,9 @@ def remove(txn):
                 ),
                 args,
             )
-            res = txn.fetchall()
+            res = cast(
+                List[Tuple[str, str, str, Optional[str], str, int, int]], txn.fetchall()
+            )
```

Comment:
Part of me wishes we were using bits of sqlalchemy so we could [get this kind of inference for free](https://docs.sqlalchemy.org/en/14/orm/extensions/mypy.html). I don't know how much you have to buy in to using the ORM for that to pay off though.

Reply:
That looks pretty handy. I wonder how hard it is to make a mypy module that would do that for us.

Reply:
I would imagine fairly hard? I'm imagining something which parses sql queries to work out what's being selected, and cross-reference against a schema to work out what type will be fed back into Python.